cmd/maya-exporter/app/collector/collector.go

```diff
@@ -0,0 +1,155 @@
+package collector
+
+import (
+	"github.com/golang/glog"
+	"github.com/openebs/maya/types/v1"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// collector inherits the properties of cstor and jiva,
+// these properties includes metrics of the volumes.
+type collector struct {
+	Volume
+	metrics
+}
+
+func New(vol Volume, casType string) *collector {
+	return &collector{
+		vol,
+		Metrics(casType),
+	}
+}
+
+// gaugeList returns the list of the registered gauge variables
+func (c *collector) gaugesList() []prometheus.Gauge {
+	return []prometheus.Gauge{
+		c.reads,
+		c.writes,
+		c.totalReadBytes,
+		c.totalWriteBytes,
+		c.totalReadTime,
+		c.totalWriteTime,
+		c.totalReadBlockCount,
+		c.totalWriteBlockCount,
+		c.actualUsed,
+		c.logicalSize,
+		c.sectorSize,
+		c.sizeOfVolume,
+		c.volumeStatus,
+	}
+}
+
+// counterList returns the list of registered counter variables
+func (c *collector) countersList() []prometheus.Collector {
+	return []prometheus.Collector{
+		c.volumeUpTime,
+		c.connectionErrorCounter,
+		c.connectionRetryCounter,
+		c.totalReplicaCounter,
+		c.degradedReplicaCounter,
+		c.healthyReplicaCounter,
+	}
+}
+
+// Describe sends the super-set of all possible descriptors of metrics
+// collected by this Collector to the provided channel and returns once
+// the last descriptor has been sent. The sent descriptors fulfill the
+// consistency and uniqueness requirements described in the Desc
+// documentation. (It is valid if one and the same Collector sends
```

Comment:
IMO an example will help understand the reader everything that has been written here as comments.

Reply:
done